crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -318,6 +318,78 @@ impl TEdgeTomlVersion {
     }
 }
 
+pub trait AppendRemoveItem {
```

Comment:
I'm wondering whether we actually do want to implement this (and the accompanying macro for implementing it) in `tedge_config_macros/src/...` as it does need to be used in so many examples.

In order to help people find the relevant place to implement the trait (as well as the macro that likely does it for them), we could use a [custom diagnostic attribute](https://blog.rust-lang.org/2024/05/02/Rust-1.78.0.html#diagnostic-attributes).

Reply:
It would also be useful to add some documentation of what this is for (i.e. making `tedge config add/remove` work)